pkgs/by-name/xp/xppen/package.nix

```diff
@@ -0,0 +1,53 @@
+{ stdenv
+, lib
+, fetchzip
+, libusb
+, autoPatchelfHook
+, qt5
+, libGL
+}:
+stdenv.mkDerivation rec {
+  pname = "xppen";
+  version = "3.4.9";
+
+  src = fetchzip {
+    extension = "tar.gz";
+    url = "https://www.xp-pen.com/download/file.html?id=1936&pid=990&ext=gz";
+    hash = "sha256-A/dv6DpelH0NHjlGj32tKv37S+9q3F8cYByiYlMuqLg=";
+  };
+
+  nativeBuildInputs = [ autoPatchelfHook ];
+
+  buildInputs = [ qt5.full libusb libGL ];
+
+  dontConfigure = true;
+  dontBuild = true;
+  dontCheck = true;
+
+  installPhase = ''
+        runHook preInstall
+        mkdir -p $out/bin $out/share
+        cp -r App/usr/lib/pentablet/. $out/bin
+        rm $out/bin/PenTablet.sh
+        cp -r App/usr/share/. $out/share
+        rm -r $out/bin/lib
+        cp -r App/lib $out/lib
+        runHook postInstall
+  '';
+
+  postInstall = ''
+        substituteInPlace $out/share/applications/xppentablet.desktop \
+          --replace "/usr/lib/pentablet/PenTablet.sh" "PenTablet" \
+          --replace "/usr/share/icons/hicolor/256x256/apps/xppentablet.png" "xppentablet"
+  '';
+
+  meta = with lib; {
+    description = "XPPen driver";
+    homepage = "https://www.xp-pen.com/";
+    downloadPage = "https://www.xp-pen.com/download/";
+    license = licenses.unfreeRedistributable;
```

Comment:
You mentioned being usure about the Licensing.
Looking into the the archive at `/App/usr/lib/doc/EULA` is an End User Agreement.
The relevant parts seem to these two:
```
1.License
This EULA grants you a non-exclusive, revocable, non-transferable, limited license to download, install and use the SOFTWARE after purchasing a Hanvon Ugee product from Hanvon Ugee or its subsidiaries and affiliates ,or through its authorized sales channels, you shall strictly abide by the terms of this EULA. 
2.Restrictions
Except for the limited license expressly granted in the "License" section of this EULA, you have no other rights to this SOFTWARE . More restrictions on your use of this SOFTWARE are as follows. You must not:
A. copy or use the SOFTWARE for any purpose other than as permitted under the above section  "License".
B. modify, make derivative works of, disassemble, decrypt, reverse compile or reverse engineer any part of the SOFTWARE.
C. remove, alter or obscure any proprietary notice (including any notice of copyright or trademark) of the author or its affiliates, partners of the SOFTWARE.
```
TLDR: You may `download, install and use the SOFTWARE`, but not redistribute.
-> Use `licenses.unfree` instead

Caveat: I am not a Lawyer and this isn't legal advise. :smile:

Reply:
> Does that just mean that there will be no source copies available in the cache?

Both this, and the fact that you have to explicitly set an envvar or config flag to allow for installing unfree packges.